Source/DafnyCore/Compilers/Java/Compiler-java.cs

```diff
@@ -2622,30 +2622,58 @@ protected class ClassWriter : IClassWriter {
       coerceE1 = false;
 
       void doPossiblyNativeBinOp(string o, string name, out string preOpS, out string opS,
-        out string postOpS, out string callS) {
+        out string postOpS, out string callS, out string staticCallS) {
         if (AsNativeType(resultType) != null) {
           var nativeName = GetNativeTypeName(AsNativeType(resultType));
-          preOpS = $"({nativeName}) {CastIfSmallNativeType(resultType)} (";
-          opS = o;
-          postOpS = ")";
-          callS = null;
+          if (o == ">>>" && resultType.NormalizeExpand() is
+                BitvectorType { Width: var width and (8 or 16 or 32 or 64) }) {
+            /// Solves https://github.com/dafny-lang/dafny/issues/3734
+            preOpS = width switch {
+              8 => "(byte)",
+              16 => "(short)",
+              32 => "(int)",
+              _ /*64*/ => "(long)"
+            };
```

Comment:
Can you use `CastIfSmallNativeType` for this? (I expect the `(int)` and `(long)` casts are identity casts anyway.)

Reply:
Looks like it works. Implemented!